docs/docs/general/03-Element Selection/recommended-practices.md

```diff
@@ -4,168 +4,242 @@ title: Recommended Practices
+await aui.click().button().withText('Sign in');
+                 .or().button().containsText('Sign')
+                 .or().customElement({
+                     customImage: 'custom_elements/signin.button.png'
+                   })
+                 .exec();
 ```
 
-## Scrolling
-When you use AskUI you can only interact with elements that you can see on your screen. Therefore you have to scroll down/sideways to interact with currently invisible elements.
+### Icon Detection Fails or Labels Wrong Icon Name
+Detecting icons correctly is difficult because there are millions of icons available. You typically run into two problems:
 
-### Scrolling on Touch Displays
-On touch displays you have to recreate the swipe gesture:
+* Icon is not detected at all
+* Icon name changes from run to run: `Search` becomes `Calendar` for example
 
-```javascript
-// Touch the display with your finger, move finger to the left, release
-await aui.mouseToggleDown().exec()
-await aui.moveMouseRelatively(-1500, 0).exec()
-await aui.mouseToggleUp().exec()
+Our recommended approach in both cases is to use a custom element instead. In case the icon is detected reliably, but only the name changes, you can alsso use a more specific element-description together with [relational selectors](../../api/04-Relations/above.md) to select the icon:
+
+```typescript
+// Custom element instead of icon
+await aui.click().customElement({
+    customImage: 'custom_elements/file.icon.png'
+  }).exec();
+
+// Make element-description more specific
+// with a relational selector
+await aui.click().icon()
+                 .leftOf().text('File')
+                 .exec();
 ```
 
-### Scrolling With Mouse Wheel
-If you want to scroll with your mouse wheel you can use the `scroll()` action:
+### Button Not Detected as Element
+Sometimes a button is not detected as one. Usually this is because of low contrast or the button lacking typical characteristics of a button.
+
+The only thing you can do is to use a custom element:
 
-```javascript
-// Scroll 10 down in y direction<>
-await aui.scroll(0, -10).exec()
+```typescript
+// Custom element instead of button().withText()
+await aui.click().customElement({
+                     customImage: 'custom_elements/signout.button.png'
+                   })
+                   .exec();
 ```
 
-### Scrolling With Key Press
-If you want to scroll with a key press you could use your arrow keys (`up`, `down`, `left`, `right` ) or the `pagedown`-key.
+---
 
-```javascript
-// Press down arrow key
-await aui.pressKey('down').exec()
+<!-- vale off -->
+## Color of Same Element Changes
+<!-- vale on -->
+AskUI can not detect color (yet). But you can use two custom elements together with `or()` to prepare for a change of color of the element:
 
-// Press up arrow key
-await aui.pressKey('up').exec()
+```typescript
+await aui.click().customElement({customImage: 'custom_elements/red.text.png'})
```

Comment:
Or you use the default grayscale of custom element which can also work in these cases.